app/src/main/java/co/touchlab/kampkit/android/BreedViewModel.kt

```diff
@@ -3,70 +3,20 @@ package co.touchlab.kampkit.android
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import co.touchlab.kampkit.db.Breed
-import co.touchlab.kampkit.injectLogger
-import co.touchlab.kampkit.models.BreedModel
-import co.touchlab.kampkit.models.DataState
-import co.touchlab.kampkit.models.ItemDataSummary
+import co.touchlab.kampkit.models.BreedCommonViewModel
+import co.touchlab.kampkit.models.BreedRepository
 import co.touchlab.kermit.Logger
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.flattenMerge
-import kotlinx.coroutines.flow.flowOf
-import kotlinx.coroutines.launch
-import org.koin.core.component.KoinComponent
 
-class BreedViewModel : ViewModel(), KoinComponent {
+class BreedViewModel(
+    breedRepository: BreedRepository,
+    log: Logger
+) : ViewModel() {
 
-    private val log: Logger by injectLogger("BreedViewModel")
-    private val scope = viewModelScope
-    private val breedModel: BreedModel = BreedModel()
-    private val _breedStateFlow: MutableStateFlow<DataState<ItemDataSummary>> = MutableStateFlow(
-        DataState(loading = true)
-    )
+    private val commonViewModel = BreedCommonViewModel(breedRepository, log, viewModelScope)
```

Comment:
I'm wondering, do we need to hide the commonViewModel and duplicate it's signature here? Is there any reason not to have this classes responsibility be just creating the common one and making it accessible to callers? I think the main reason would be confusion at the call site `breedViewModel.commonViewModel.refreshBreeds()` is obviously pretty ugly. But with some improved naming, might be worth it to avoid maintaining the duplicated signatures?

Reply:
In KaMPKit we don't, which is a refactor on my (undocumented, oops) todo list before merging this. In general though, you might want this separation if your platforms are not perfectly in-sync because it gives you a place to put android-specific stuff. It's also a place you can convert to LiveData or RxJava or anything else platform-specific you might be using in existing Android code.